### PR TITLE
RLM-895 Remove old Kilo and Liberty jobs

### PR DESCRIPTION
**Pull request by claco**
This change removes the old style rpc-openstack jobs in rpc_aio.yml
that rarely pass and are covered elsewhere in rpc_openstack.yml using
the new gating methodology.

This change removes the Kilo/Liberty jobs from rpc_openstack.yml,
which are old, not maintained, and are somewhat redundant as we test
Kilo/Liberty installations as part of rpc_upgrades.yml.

This will reduce job duplication, and job failure reporting that the
triage team currently investigates for no good reason as Kilo/Liberty
are EOL.

This change is meant to remove:
- jenkins-PM_rpc-openstack-kilo-trusty-ceph-deploy
- jenkins-PM_rpc-openstack-kilo-trusty-swift-deploy
- jenkins-PM_rpc-openstack-liberty-trusty-ceph-deploy
- jenkins-PM_rpc-openstack-liberty-trusty-swift-deploy

Issue: [RLM-895](https://rpc-openstack.atlassian.net/browse/RLM-895)

### CODE REVIEW

**Comment by claco** [comment]
For reference, the jobs named are in this triage queue:  https://rpc-openstack.atlassian.net/issues/?filter=13955

Search for kilo or liberty.

File: rpc_jobs/rpc_openstack.yml [hunk 1]
```diff
@@ -142,93 +142,6 @@
     jobs:
       - 'PR_{repo_name}-{series}-{image}-{scenario}-{action}'
 
-- project:
-    name: "rpc-openstack-liberty-premerge"
```
Comment (odyssey4me) [inline rpc_jobs/rpc_openstack.yml]: This is a PR test, not a PM test. Are you sure you want to remove PR tests too?

File: rpc_jobs/rpc_openstack.yml [hunk 2]
```diff
@@ -142,93 +142,6 @@
...
-      - "ceph"
-      - "swift"
-    action:
-      - deploy:
-          FLAVOR: "performance2-15"
-    jobs:
-      - 'PR_{repo_name}-{series}-{image}-{scenario}-{action}'
-
-- project:
-    name: "rpc-openstack-liberty-tox-premerge"
```
Comment (odyssey4me) [inline rpc_jobs/rpc_openstack.yml]: This is a PR test, not a PM test. Are you sure you want to remove PR tests too?

File: rpc_jobs/rpc_openstack.yml [hunk 3]
```diff
@@ -142,93 +142,6 @@
...
-    scenario:
-      - "ansible-lint"
-      - "flake8"
-    action:
-      - "tox-test"
-    jobs:
-      - 'PR_{repo_name}-{series}-{image}-{scenario}-{action}'
-
-- project:
-    name: "rpc-openstack-kilo-premerge"
```
Comment (odyssey4me) [inline rpc_jobs/rpc_openstack.yml]: This is a PR test, not a PM test. Are you sure you want to remove PR tests too?

File: rpc_jobs/rpc_openstack.yml [hunk 4]
```diff
@@ -142,93 +142,6 @@
...
-    scenario:
-      - "swift"
-    action:
-      - deploy:
-          FLAVOR: "performance2-15"
-    jobs:
-      - 'PR_{repo_name}-{series}-{image}-{scenario}-{action}'
-
-- project:
-    name: "rpc-openstack-kilo-tox-premerge"
```
Comment (odyssey4me) [inline rpc_jobs/rpc_openstack.yml]: This is a PR test, not a PM test. Are you sure you want to remove PR tests too?

File: rpc_jobs/rpc_openstack.yml (hunk 1 above)
Comment (claco) [reply rpc_jobs/rpc_openstack.yml]: I'm not sure on the outcome, but it was a deliberate decision. No one really should be committing things to Liberty anymore other than the ones we've been doing to get these old gates to pass.

The last official release was Nov 10, 2016.

File: rpc_jobs/rpc_openstack.yml (hunk 2 above)
Comment (claco) [reply rpc_jobs/rpc_openstack.yml]: I'm not sure on the outcome, but it was a deliberate decision. No one really should be committing things to Liberty anymore other than the ones we've been doing to get these old gates to pass.

The last official release was Nov 10, 2016.

File: rpc_jobs/rpc_openstack.yml (hunk 3 above)
Comment (claco) [reply rpc_jobs/rpc_openstack.yml]: Yup. See above. Kilo isn't even installable without hacked workarounds using sed/awk, etc. We've encapsulated those in rpc-upgrade for gating purposes. No one should be installing Kilo outside of that.

File: rpc_jobs/rpc_openstack.yml (hunk 4 above)
Comment (claco) [reply rpc_jobs/rpc_openstack.yml]: Same as previous. Thanks for double checking though!

**Pull request merged**